In file included from ../src/bufferutil.cc:15:0:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「void NanThrowError(const char*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:209:7: 錯誤：「ThrowException」 不是 「v8」 的成員
       v8::ThrowException(fun(v8::String::New(errmsg)));                        \
       ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:213:5: 附註：in expansion of macro 「_NAN_THROW_ERROR」
     _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:209:30: 錯誤：「New」 不是 「v8::String」 的成員
       v8::ThrowException(fun(v8::String::New(errmsg)));                        \
                              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:213:5: 附註：in expansion of macro 「_NAN_THROW_ERROR」
     _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「void NanThrowError(v8::Handle<v8::Value>)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:218:5: 錯誤：「ThrowException」 不是 「v8」 的成員
     v8::ThrowException(error);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「void NanThrowError(const char*, int)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:222:53: 錯誤：「New」 不是 「v8::String」 的成員
     v8::Local<v8::Value> err = v8::Exception::Error(v8::String::New(msg));
                                                     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:224:14: 錯誤：「New」 不是 「v8::String」 的成員
     obj->Set(v8::String::New("code"), v8::Int32::New(errorNumber));
              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:224:65: 錯誤：對 「v8::Int32::New(const int&)」 的呼叫沒有匹配的函式
     obj->Set(v8::String::New("code"), v8::Int32::New(errorNumber));
                                                                 ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:224:65: 附註：candidate is:
In file included from ../src/bufferutil.cc:7:0:
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:2012:25: 附註：static v8::Local<v8::Integer> v8::Integer::New(v8::Isolate*, int32_t)
   static Local<Integer> New(Isolate* isolate, int32_t value);
                         ^
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:2012:25: 附註：  candidate expects 2 arguments, 1 provided
In file included from ../src/bufferutil.cc:15:0:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「void NanThrowTypeError(const char*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:209:7: 錯誤：「ThrowException」 不是 「v8」 的成員
       v8::ThrowException(fun(v8::String::New(errmsg)));                        \
       ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:229:5: 附註：in expansion of macro 「_NAN_THROW_ERROR」
     _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:209:30: 錯誤：「New」 不是 「v8::String」 的成員
       v8::ThrowException(fun(v8::String::New(errmsg)));                        \
                              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:229:5: 附註：in expansion of macro 「_NAN_THROW_ERROR」
     _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「void NanThrowRangeError(const char*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:209:7: 錯誤：「ThrowException」 不是 「v8」 的成員
       v8::ThrowException(fun(v8::String::New(errmsg)));                        \
       ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:233:5: 附註：in expansion of macro 「_NAN_THROW_ERROR」
     _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:209:30: 錯誤：「New」 不是 「v8::String」 的成員
       v8::ThrowException(fun(v8::String::New(errmsg)));                        \
                              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:233:5: 附註：in expansion of macro 「_NAN_THROW_ERROR」
     _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In constructor 「NanCallback::NanCallback(const v8::Local<v8::Function>&)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:450:49: 錯誤：對 「v8::Object::New()」 的呼叫沒有匹配的函式
     v8::Local<v8::Object> obj = v8::Object::New();
                                                 ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:450:49: 附註：candidate is:
In file included from ../src/bufferutil.cc:7:0:
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:2388:24: 附註：static v8::Local<v8::Object> v8::Object::New(v8::Isolate*)
   static Local<Object> New(Isolate* isolate);
                        ^
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:2388:24: 附註：  candidate expects 1 argument, 0 provided
In file included from ../src/bufferutil.cc:15:0:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:103:26: 錯誤：「NewSymbol」 不是 「v8::String」 的成員
 #define NanSymbol(value) v8::String::NewSymbol(value)
                          ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:451:14: 附註：in expansion of macro 「NanSymbol」
     obj->Set(NanSymbol("callback"), fn);
              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In destructor 「NanCallback::~NanCallback()」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:457:12: 錯誤：「class v8::Persistent<v8::Object>」 沒具名為 「Dispose」 的成員
     handle.Dispose();
            ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:458:12: 錯誤：「class v8::Persistent<v8::Object>」 沒具名為 「Clear」 的成員
     handle.Clear();
            ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In member function 「v8::Local<v8::Function> NanCallback::GetFunction()」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:103:26: 錯誤：「NewSymbol」 不是 「v8::String」 的成員
 #define NanSymbol(value) v8::String::NewSymbol(value)
                          ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:462:46: 附註：in expansion of macro 「NanSymbol」
     return NanPersistentToLocal(handle)->Get(NanSymbol("callback"))
                                              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:463:25: 錯誤：expected primary-expression before 「>」 token
         .As<v8::Function>();
                         ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:463:27: 錯誤：expected primary-expression before 「)」 token
         .As<v8::Function>();
                           ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In member function 「void NanCallback::Call(int, v8::Local<v8::Value>*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:103:26: 錯誤：「NewSymbol」 不是 「v8::String」 的成員
 #define NanSymbol(value) v8::String::NewSymbol(value)
                          ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:475:12: 附註：in expansion of macro 「NanSymbol」
        Get(NanSymbol("callback")).As<v8::Function>();
            ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:475:50: 錯誤：expected primary-expression before 「>」 token
        Get(NanSymbol("callback")).As<v8::Function>();
                                                  ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:475:52: 錯誤：expected primary-expression before 「)」 token
        Get(NanSymbol("callback")).As<v8::Function>();
                                                    ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:477:20: 錯誤：「GetCurrent」 不是 「v8::Context」 的成員
     callback->Call(v8::Context::GetCurrent()->Global(), argc, argv);
                    ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: At global scope:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:518:3: 錯誤：「uv_work_t」 沒有命名一個類型
   uv_work_t request;
   ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In constructor 「NanAsyncWorker::NanAsyncWorker(NanCallback*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:490:5: 錯誤：「request」 在此作用欄位中尚未宣告
     request.data = this;
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In member function 「void NanAsyncWorker::SavePersistent(const char*, v8::Local<v8::Object>&)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:103:26: 錯誤：「NewSymbol」 不是 「v8::String」 的成員
 #define NanSymbol(value) v8::String::NewSymbol(value)
                          ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:529:17: 附註：in expansion of macro 「NanSymbol」
     handle->Set(NanSymbol(key), obj);
                 ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In member function 「v8::Local<v8::Object> NanAsyncWorker::GetFromPersistent(const char*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:103:26: 錯誤：「NewSymbol」 不是 「v8::String」 的成員
 #define NanSymbol(value) v8::String::NewSymbol(value)
                          ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:536:24: 附註：in expansion of macro 「NanSymbol」
     return handle->Get(NanSymbol(key)).As<v8::Object>();
                        ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:536:53: 錯誤：expected primary-expression before 「>」 token
     return handle->Get(NanSymbol(key)).As<v8::Object>();
                                                     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:536:55: 錯誤：expected primary-expression before 「)」 token
     return handle->Get(NanSymbol(key)).As<v8::Object>();
                                                       ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In member function 「virtual void NanAsyncWorker::HandleErrorCallback()」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:549:30: 錯誤：「New」 不是 「v8::String」 的成員
         v8::Exception::Error(v8::String::New(errmsg))
                              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: At global scope:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:555:30: 錯誤：變數或欄位 「NanAsyncExecute」 宣告為 void
 inline void NanAsyncExecute (uv_work_t* req) {
                              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:555:30: 錯誤：「uv_work_t」 在此作用欄位中尚未宣告
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:555:41: 錯誤：「req」 在此作用欄位中尚未宣告
 inline void NanAsyncExecute (uv_work_t* req) {
                                         ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:560:38: 錯誤：變數或欄位 「NanAsyncExecuteComplete」 宣告為 void
 inline void NanAsyncExecuteComplete (uv_work_t* req) {
                                      ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:560:38: 錯誤：「uv_work_t」 在此作用欄位中尚未宣告
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:560:49: 錯誤：「req」 在此作用欄位中尚未宣告
 inline void NanAsyncExecuteComplete (uv_work_t* req) {
                                                 ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「void NanAsyncQueueWorker(NanAsyncWorker*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:568:23: 錯誤：「uv_default_loop」 在此作用欄位中尚未宣告
       uv_default_loop()
                       ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:569:16: 錯誤：「class NanAsyncWorker」 沒具名為 「request」 的成員
     , &worker->request
                ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:570:7: 錯誤：「NanAsyncExecute」 在此作用欄位中尚未宣告
     , NanAsyncExecute
       ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:571:8: 錯誤：「uv_after_work_cb」 在此作用欄位中尚未宣告
     , (uv_after_work_cb)NanAsyncExecuteComplete
        ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:572:3: 錯誤：「uv_queue_work」 在此作用欄位中尚未宣告
   );
   ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「bool _NanGetExternalParts(v8::Handle<v8::Value>, const char**, size_t*)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:716:78: 錯誤：對 「v8::Local<v8::String>::New(v8::Handle<v8::String>)」 的呼叫沒有匹配的函式
   v8::Local<v8::String> str = v8::Local<v8::String>::New(val.As<v8::String>());
                                                                              ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:716:78: 附註：candidates are:
In file included from ../src/bufferutil.cc:7:0:
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:368:29: 附註：static v8::Local<T> v8::Local<T>::New(v8::Isolate*, v8::Handle<T>) [with T = v8::String]
   V8_INLINE static Local<T> New(Isolate* isolate, Handle<T> that);
                             ^
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:368:29: 附註：  candidate expects 2 arguments, 1 provided
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:369:29: 附註：static v8::Local<T> v8::Local<T>::New(v8::Isolate*, const v8::PersistentBase<T>&) [with T = v8::String]
   V8_INLINE static Local<T> New(Isolate* isolate,
                             ^
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:369:29: 附註：  candidate expects 2 arguments, 1 provided
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:391:29: 附註：static v8::Local<T> v8::Local<T>::New(v8::Isolate*, T*) [with T = v8::String]
   V8_INLINE static Local<T> New(Isolate* isolate, T* that);
                             ^
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:391:29: 附註：  candidate expects 2 arguments, 1 provided
In file included from ../src/bufferutil.cc:15:0:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In function 「char* NanFromV8String(v8::Handle<v8::Value>, Nan::Encoding, size_t*, char*, size_t, int)」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:769:3: 錯誤：「AsciiValue」 不是 「v8::String」 的成員
   v8::String::AsciiValue value(toStr);
   ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:769:26: 錯誤：expected 「;」 before 「value」
   v8::String::AsciiValue value(toStr);
                          ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:832:39: 錯誤：「value」 在此作用欄位中尚未宣告
       sz_ = _nan_base64_decoded_size(*value, toStr->Length());
                                       ^
../src/bufferutil.cc: In static member function 「static void BufferUtil::Initialize(v8::Handle<v8::Object>)」:
../src/bufferutil.cc:27:58: 錯誤：對 「v8::FunctionTemplate::New(void (&)(const v8::FunctionCallbackInfo<v8::Value>&))」 的呼叫沒有匹配的函式
     Local<FunctionTemplate> t = FunctionTemplate::New(New);
                                                          ^
../src/bufferutil.cc:27:58: 附註：candidate is:
In file included from ../src/bufferutil.cc:7:0:
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:3455:34: 附註：static v8::Local<v8::FunctionTemplate> v8::FunctionTemplate::New(v8::Isolate*, v8::FunctionCallback, v8::Handle<v8::Value>, v8::Handle<v8::Signature>, int)
   static Local<FunctionTemplate> New(
                                  ^
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:3455:34: 附註：  no known conversion for argument 1 from 「void(const v8::FunctionCallbackInfo<v8::Value>&)」 to 「v8::Isolate*」
../src/bufferutil.cc:32:17: 錯誤：「NewSymbol」 不是 「v8::String」 的成員
     target->Set(String::NewSymbol("BufferUtil"), t->GetFunction());
                 ^
In file included from ../src/bufferutil.cc:15:0:
../src/bufferutil.cc: In static member function 「static void BufferUtil::Merge(const v8::FunctionCallbackInfo<v8::Value>&)」:
../src/bufferutil.cc:60:25: 錯誤：too few arguments to function 「v8::Handle<v8::Boolean> v8::True(v8::Isolate*)」
     NanReturnValue(True());
                         ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:195:65: 附註：in definition of macro 「NanReturnValue」
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                 ^
In file included from ../src/bufferutil.cc:7:0:
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:6613:17: 附註：declared here
 Handle<Boolean> True(Isolate* isolate) {
                 ^
In file included from ../src/bufferutil.cc:15:0:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:195:70: 錯誤：在回傳「void」的函式中，回傳敘述帶回傳值 [-fpermissive]
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                      ^
../src/bufferutil.cc:60:5: 附註：in expansion of macro 「NanReturnValue」
     NanReturnValue(True());
     ^
../src/bufferutil.cc: In static member function 「static void BufferUtil::Unmask(const v8::FunctionCallbackInfo<v8::Value>&)」:
../src/bufferutil.cc:81:25: 錯誤：too few arguments to function 「v8::Handle<v8::Boolean> v8::True(v8::Isolate*)」
     NanReturnValue(True());
                         ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:195:65: 附註：in definition of macro 「NanReturnValue」
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                 ^
In file included from ../src/bufferutil.cc:7:0:
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:6613:17: 附註：declared here
 Handle<Boolean> True(Isolate* isolate) {
                 ^
In file included from ../src/bufferutil.cc:15:0:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:195:70: 錯誤：在回傳「void」的函式中，回傳敘述帶回傳值 [-fpermissive]
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                      ^
../src/bufferutil.cc:81:5: 附註：in expansion of macro 「NanReturnValue」
     NanReturnValue(True());
     ^
../src/bufferutil.cc: In static member function 「static void BufferUtil::Mask(const v8::FunctionCallbackInfo<v8::Value>&)」:
../src/bufferutil.cc:106:25: 錯誤：too few arguments to function 「v8::Handle<v8::Boolean> v8::True(v8::Isolate*)」
     NanReturnValue(True());
                         ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:195:65: 附註：in definition of macro 「NanReturnValue」
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                 ^
In file included from ../src/bufferutil.cc:7:0:
/home/aiderhuang/.node-gyp/0.12.3/deps/v8/include/v8.h:6613:17: 附註：declared here
 Handle<Boolean> True(Isolate* isolate) {
                 ^
In file included from ../src/bufferutil.cc:15:0:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:195:70: 錯誤：在回傳「void」的函式中，回傳敘述帶回傳值 [-fpermissive]
 # define NanReturnValue(value) return args.GetReturnValue().Set(value)
                                                                      ^
../src/bufferutil.cc:106:5: 附註：in expansion of macro 「NanReturnValue」
     NanReturnValue(True());
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h: In instantiation of 「void NanDispose(v8::Persistent<T, v8::NonCopyablePersistentTraits<T> >&) [with T = v8::Object]」:
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:498:34:   required from here
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:237:5: 錯誤：「class v8::Persistent<v8::Object>」 沒具名為 「Dispose」 的成員
     handle.Dispose(nan_isolate);
     ^
/home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws/node_modules/nan/nan.h:238:5: 錯誤：「class v8::Persistent<v8::Object>」 沒具名為 「Clear」 的成員
     handle.Clear();
     ^
make: *** [Release/obj.target/bufferutil/src/bufferutil.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Linux 3.10.0-229.1.2.el7.x86_64
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/aiderhuang/nodejs/UDID/node_modules/socket.io/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/ws
gyp ERR! node -v v0.12.3
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok 
